src/many-ledger/src/module/ledger_tokens.rs

```diff
@@ -0,0 +1,197 @@
+
+        if let Some(Either::Left(addr)) = &args.owner {
+            verify_tokens_acl(&self.storage, sender, addr, [Role::CanTokensCreate])?;
+        }
+
+        let ticker = &args.summary.ticker;
+        if self
+            .storage
+            .get_symbols_and_tickers()?
+            .values()
+            .any(|v| v == ticker)
+        {
+            return Err(ManyError::unknown(format!(
+                "The ticker {ticker} already exists on this network"
+            )));
+        }
+        self.storage.create_token(sender, args)
+    }
+
+    fn info(&self, _sender: &Address, args: TokenInfoArgs) -> Result<TokenInfoReturns, ManyError> {
+        // Check the memory symbol cache for requested symbol
+        if !self.storage.migrations().is_active(&TOKEN_MIGRATION) {
+            return Err(ManyError::unknown(
+                "Token Migration needs to be active to use this endpoint",
+            ));
+        }
+
+        let symbol = &args.symbol;
+        if !self.storage.get_symbols()?.contains(symbol) {
+            return Err(ManyError::unknown(format!(
+                "The symbol {symbol} was not found"
+            )));
+        }
+        self.storage.info_token(args)
+    }
+
+    fn update(
+        &mut self,
+        sender: &Address,
+        args: TokenUpdateArgs,
+    ) -> Result<TokenUpdateReturns, ManyError> {
+        // TODO: Limit token update to given sender
+        // | A server implementing this attribute SHOULD protect the endpoints described in this form in some way.
+        // | For example, endpoints SHOULD error if the sender isn't from a certain address.
+
+        if !self.storage.migrations().is_active(&TOKEN_MIGRATION) {
+            return Err(ManyError::unknown(
+                "Token Migration needs to be active to use this endpoint",
+            ));
+        }
+
+        // Get the current owner and check if we're allowed to update this token
+        let current_owner = self.storage.get_owner(&args.symbol)?;
+        match current_owner {
+            Some(addr) => {
+                verify_tokens_acl(&self.storage, sender, &addr, [Role::CanTokensUpdate])?;
+            }
+            None => {
+                return Err(ManyError::unknown(
+                    "Unable to update, this token is immutable",
+                ))
+            }
+        }
+
+        // Check the memory symbol cache for requested symbol
+        let symbol = &args.symbol;
+        if !self.storage.get_symbols()?.contains(symbol) {
+            return Err(ManyError::unknown(format!(
+                "The symbol {symbol} was not found"
+            )));
+        }
+
+        self.storage.update_token(sender, args)
+    }
+
+    fn add_extended_info(
+        &mut self,
+        sender: &Address,
+        args: TokenAddExtendedInfoArgs,
+    ) -> Result<TokenAddExtendedInfoReturns, ManyError> {
+        // TODO: Limit adding extended info to given sender
```

Comment:
Same as above. Maybe only creation should be limited.

Reply:
Limit creation to MFX address